src/main/java/com/github/manolo8/darkbot/config/ConfigManager.java

```diff
@@ -103,22 +107,34 @@ private Config loadConfig(Path configFile, Path backupFile) {
     public IDarkBotAPI getAPI(StartupParams params) {
         if (params.useNoOp()) return new NoopApiAdapter();
         try {
+
             if (config.BOT_SETTINGS.API_CONFIG.API == 0) return new DarkBotApiAdapter();
             else if (config.BOT_SETTINGS.API_CONFIG.API == 1) return new DarkFlashApiAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 2) return new DarkBoatAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 3) return new NativeApiAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 4) return new NoopApiAdapter();
             else throw new IllegalArgumentException("API not found: " + config.BOT_SETTINGS.API_CONFIG.API);
         } catch (Error e) {
-            System.out.println("Error enabling API #" + config.BOT_SETTINGS.API_CONFIG.API + ", using no-op api");
             e.printStackTrace();
-            config.BOT_SETTINGS.API_CONFIG.API = 4;
-            Popups.showMessageAsync(
-                    "API failed to load",
-                    "The API you had selected is not able to load.\n" +
-                    "You probably do not have the required DLL in your lib folder.\n" +
-                    "The bot will start on no-operation API, change it in the settings and restart.",
-                    JOptionPane.ERROR_MESSAGE);
+
+            APIErrors err = new APIErrors();
```

Comment:
An instance of APIErrors doesn't feel justified here whatsoever, make the class have static methods.

I think the only thing you should need is something along the lines of:

`String description = ApiErrors.createDescription(e)`

You can append that to the end of the original message, you can move it to the resources too and make it html in itself, with a parameter being the description.

Could end up looking something like:

```
The API you had selected is not able to load.
You probably do not have the required DLL in your lib folder.
The bot will start on no-operation API, change it in the settings and restart

In this case, it seems like you may be missing runtimes, download them here:
https://... 
```
The last part of that message would depend on what you find to be the issue